labextension/src/lib/CellUtils.ts

```diff
@@ -15,11 +17,13 @@ export default class CellUtilities {
    * @description Reads the output at a cell within the specified notebook and returns it as a string
    * @param notebook The notebook to get the cell from
    * @param index The index of the cell to read
-   * @returns any - A string value of the cell output from the specified
+   * @returns MultilineString | PartialJSONObject | null | undefined - A string value of the cell output from the specified
    * notebook and cell index, or null if there is no output.
    * @throws An error message if there are issues in getting the output
    */
-  public static readOutput(notebook: Notebook, index: number): any {
+  // here I think any type could be left because the return types are so messy 
```

Comment:
If you check the only place where this method is used the output is actually cast to `string`:

```
const output = CellUtilities.readOutput(notebook, index);
notebook.activeCellIndex = oldIndex;
return output as string;
```

Here's what I suggest:

* modify the method to return `string | undefined`
* on `return null` change to `return undefined`
* cast the returned values to string:

```
    if (out && isExecuteResult(out)) {
      return out.data['text/plain'] as string;
    }
    if (out && isStream(out)) {
      return out.text as string;
    }
```
This is not ideal, but will decrease the complexity, later we can revisit this code to make sure that it is up-to-date with later Jupyter Lab APIs.